File: alumni-back/src/main/java/ec/edu/ista/springgc1/model/dto/AnswerSearchDTO.java
package ec.edu.ista.springgc1.model.dto;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonView;

import ec.edu.ista.springgc1.model.entity.Survey;
import ec.edu.ista.springgc1.view.View;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AnswerSearchDTO implements Serializable {
	@JsonView(View.Public.class)
	 private Long id;
	@JsonView(View.Public.class)
	    private String graduadoEmail;
	@JsonView(View.Public.class)
	    private String carreraNombre;
	@JsonView(View.Public.class)
	    private String surveyTitle;
	@JsonView(View.Public.class)
	    private Map<Long, String> questionResponses;
	@JsonView(View.Public.class)
	    private String openAnswer;

}

File: alumni_app/src/app/data/model/DTO/AnswerSearchDTO.ts
export class AnswerSearchDTO {
    id?: number;
    graduadoEmail: string;
    carreraNombre: string;
    surveyTitle: string;
    questionResponses: Record<number, string>;
    openAnswer?: string;
  
    constructor(
      id: number,
      graduadoEmail: string,
      carreraNombre: string,
      surveyTitle: string,
      questionResponses: Record<number, string>,
      openAnswer: string
    ) {
      this.id = id;
      this.graduadoEmail = graduadoEmail;
      this.carreraNombre = carreraNombre;
      this.surveyTitle = surveyTitle;
      this.questionResponses = questionResponses;
      this.openAnswer = openAnswer;
    }

    assignResponseToQuestion(questionId: number, response: string): void {
      this.questionResponses[questionId] = response;
    }
  }